arrow/arrow.py

```diff
@@ -79,10 +90,22 @@ def __init__(
         elif util.isstr(tzinfo):
             tzinfo = parser.TzinfoParser.parse(tzinfo)
 
-        self._datetime = datetime(
-            year, month, day, hour, minute, second, microsecond, tzinfo
+        self._fold = fold
+
+        self._datetime = enfold(
+            datetime(year, month, day, hour, minute, second, microsecond, tzinfo),
+            fold=self._fold,
         )
 
+        # warn user if dt is imaginary
+        if not dateutil_tz.datetime_exists(self._datetime):
+            warnings.warn(
+                "{} does not exist in the current timezone - {}".format(
+                    self._datetime, tzinfo
+                ),
+                util.ImaginaryDatetimeWarning,
```

Comment:
would be nice to have this exposed as a property or variable instead of just a warning.

Reply:
Good idea, I think using a property is the way to go.